instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/decorator/BaseDecorator.java

```diff
@@ -29,19 +32,26 @@
 import java.lang.reflect.Method;
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
-import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ExecutionException;
 
 @Deprecated
 public abstract class BaseDecorator {
 
-  private static final ClassValue<SpanNames> SPAN_NAMES =
-      new ClassValue<SpanNames>() {
-        @Override
-        protected SpanNames computeValue(Class<?> type) {
-          return new SpanNames(getClassName(type));
-        }
-      };
+  private static final QualifiedClassNameCache CLASS_NAMES =
```

Comment:
Why do we need this cache at all? Is it that expensive to call `class.getSimpleName()`?

Reply:
Well, it does slightly more than that. No, it's not expensive to call `class.getSimpleName()` - but we do repeatedly concatenate class names with method names to create resource names, and it adds up. We found these strings tend to get promoted when spans linger. It may not be necessary in this fork.